rust/src/measurements/make_private_lazyframe/aggregate/mod.rs

```diff
@@ -0,0 +1,189 @@
+use std::collections::BTreeSet;
+
+use crate::combinators::{make_basic_composition, BasicCompositionMeasure};
+use crate::core::{Function, Measurement, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{ExprContext, ExprDomain, LogicalPlanDomain};
+use crate::error::*;
+use crate::measurements::make_private_expr;
+use crate::metrics::PartitionDistance;
+use crate::traits::InfMul;
+use crate::transformations::traits::UnboundedMetric;
+use crate::transformations::{DatasetMetric, StableLogicalPlan};
+use make_private_expr::PrivateExpr;
+use polars::prelude::*;
+use polars_plan::prelude::GroupbyOptions;
+
+/// Create a private version of an aggregate operation on a LazyFrame.
+///
+/// # Arguments
+/// * `input_domain` - The domain of the input LazyFrame.
+/// * `input_metric` - The metric of the input LazyFrame.
+/// * `output_measure` - The measure of the output LazyFrame.
+/// * `param` - The parameter for the transformation.
+/// * `plan` - The LazyFrame to transform.
+pub fn make_private_aggregate<MS, MI, MO>(
+    input_domain: LogicalPlanDomain,
+    input_metric: MS,
+    output_measure: MO,
+    param: f64,
+    plan: LogicalPlan,
+) -> Fallible<Measurement<LogicalPlanDomain, LogicalPlan, MS, MO>>
+where
+    MS: 'static + DatasetMetric,
+    MI: 'static + UnboundedMetric,
+    MO: 'static + BasicCompositionMeasure,
+    Expr: PrivateExpr<PartitionDistance<MI>, MO>,
+    LogicalPlan: StableLogicalPlan<MS, MI>,
+    (LogicalPlanDomain, MS): MetricSpace,
+    (LogicalPlanDomain, MI): MetricSpace,
+    (ExprDomain, MI): MetricSpace,
+    (ExprDomain, PartitionDistance<MI>): MetricSpace,
+{
+    let LogicalPlan::Aggregate {
+        input,
+        keys,
+        aggs,
+        apply,
+        options,
+        ..
+    } = plan.clone()
+    else {
+        return fallible!(MakeMeasurement, "Expected Aggregate logical plan");
+    };
+
+    let t_prior = input.make_stable(input_domain.clone(), input_metric.clone())?;
+    let (input_domain, input_metric) = t_prior.output_space();
+
+    if options.as_ref() != &GroupbyOptions::default() {
+        return fallible!(MakeMeasurement, "Unsupported options in logical plan");
```

Comment:
I'm not sure what `options` looks like, but if it has a string representation, would it be worth including in the error message, just to give the user more of a hint?